### PR TITLE
Fix `edition` for main crates in test compilation in test collector

### PR DESCRIPTION
**Pull request by piotmag769**

### CODE REVIEW

File: extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs [hunk 1]
```diff
@@ -180,13 +188,19 @@ fn insert_lib_entrypoint_content_into_db(
...
-            crate_root.to_path_buf(),
-        ))),
+        Some(CrateConfiguration {
+            root: Directory::Real(crate_root.to_path_buf()),
+            settings: CrateSettings {
+                edition: main_crate_edition,
+                cfg_set: None,
+                experimental_features: Default::default(),
+            },
+        }),
```
Comment (piotmag769) [inline extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs:203]: This is the actual fix

File: extensions/scarb-snforge-test-collector/src/metadata.rs [hunk 2]
```diff
@@ -150,11 +167,29 @@ impl CompilationUnit<'_> {
...
-        self.unit_metadata.target.source_root().to_path_buf()
+    pub fn main_package_source_root(&self) -> Utf8PathBuf {
+        self.main_package_metadata.source_root().to_path_buf()
     }
 
-    pub fn source_file_path(&self) -> &Utf8Path {
-        &self.unit_metadata.target.source_path
+    pub fn main_package_source_file_path(&self) -> &Utf8Path {
+        &self.main_package_metadata.source_path
+    }
```
Comment (piotmag769) [inline extensions/scarb-snforge-test-collector/src/metadata.rs:176]: Changed it bcs we actually want the source root and source file path of the package we are compiling the tests for, not of the target. They are the same for now bcs we take the CU with main package being the package we want to compile tests for, but the logic it is less prone to break in the future this way.
cc @maciektr to check if I'm right

Comment (maciektr) [reply extensions/scarb-snforge-test-collector/src/metadata.rs:176]: Ok, if you don't fully integrate with the Scarb data model it kind of makes sense.

File: extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs [hunk 3]
```diff
@@ -180,13 +188,19 @@ fn insert_lib_entrypoint_content_into_db(
...
         main_crate_id,
-        Some(CrateConfiguration::default_for_root(Directory::Real(
-            crate_root.to_path_buf(),
-        ))),
+        Some(CrateConfiguration {
+            root: Directory::Real(crate_root.to_path_buf()),
+            settings: CrateSettings {
+                edition: main_crate_edition,
+                cfg_set: None,
+                experimental_features: Default::default(),
```
[the remark this answers is not in the document]
Comment (piotmag769) [reply extensions/scarb-snforge-test-collector/src/compilation/test_collector.rs]: Done

**Review by Arcticae — APPROVED** [review]

**Pull request merged**